appshell/appshell_extensions_win.cpp

```diff
@@ -2005,6 +2009,129 @@ void DragWindow(CefRefPtr<CefBrowser> browser) {
     HWND browserHwnd = (HWND)getMenuParent(browser);
     SendMessage(browserHwnd, WM_NCLBUTTONDOWN, HTCAPTION, 0);
 }
-    
 
+// Courtesy: Rafael.
+// https://oroboro.com/unique-machine-fingerprint/
+
+// we just need this for purposes of unique machine id. 
+// So any one or two mac's is fine.
+u16 HasMacAddress(PIP_ADAPTER_INFO info)
+{
+	u16 hash = 0;
+	for (u32 i = 0; i < info->AddressLength; i++)
+	{
+		hash += (info->Address[i] << ((i & 1) * 8));
+	}
+	return hash;
+}
+
+void GetMacHash(u16& mac1, u16& mac2)
+{
+	IP_ADAPTER_INFO AdapterInfo[32];
+	DWORD dwBufLen = sizeof(AdapterInfo);
+
+	DWORD dwStatus = GetAdaptersInfo(AdapterInfo, &dwBufLen);
```

Comment:
Should you consider MIB_IF_TYPE_ETHERNET as listed in the documentation https://msdn.microsoft.com/en-us/library/windows/desktop/aa365917(v=vs.85).aspx